doc/gearbox/gearbox_mgr_design.md

```diff
@@ -0,0 +1,624 @@
+| line_adver_asym_pause  | boolean  | Line-side advertised asymmetric pause mode ```true or false```                                 |
+| line_adver_media_type  | string   | Line-side advertised media type ```"not present", "unknown", "fiber", "copper", "backplane"``` |
+
+## 3.4 DB Changes
+
+### 3.4.1 CONFIG DB
+No gearbox specific configurations are necessary at this time.
+
+### 3.4.2 APP DB
+A new GEARBOX_TABLE has been added to the application database for the purpose of storing the related gearbox
+platform and PHY related configuration parameters. This table is populated by the newly modified PORTSORCH.
+
+#### 3.4.2.1 GEARBOX_TABLE - Platform Instance
+
+REDIS example:
+~~~
+redis-dump -d 0 -k "_GEARBOX_TABLE:phy:0" -y
+{
+  "_GEARBOX_TABLE:phy:0": {
+    "type": "hash",
+    "value": {
+      "address": "0x1000",
+      "bus_id": "0",
+      "config_file": "/usr/share/sonic/hwsku/sesto-1.json",
+      "firmware_path": "/tmp/phy-sesto-1.bin",
+      "lib_name": "libsai_phy_sesto-1.so",
+      "name": "sesto-1",
+      "phy_access": "",
+      "phy_id": "0",
+      "phy_oid": "1234",
+      "sai_init_config_file": "/usr/share/sonic/hwsku/sesto-1.bcm"
+    }
+  }
+~~~
+
+#### 3.4.2.2 GEARBOX_TABLE - Platform Host Interfaces
+
+REDIS example:
+~~~
+redis-dump -d 0 -k "_GEARBOX_TABLE:interface:49" -y
+{
+  "_GEARBOX_TABLE:interface:49": {
+    "type": "hash",
+    "value": {
+      "index": "49",
+      "line_lanes": "204,205",
+      "phy_id": "0",
+      "system_lanes": "200,201,202,203"
+    }
+  }
+}
+~~~
+
+#### 3.4.2.3 GEARBOX_TABLE - Platform PHY Lanes
+
+REDIS Example:
+~~~
+redis-dump -d 0 -k "_GEARBOX_TABLE:phy:0:lanes:200" -y
+{
+  "_GEARBOX_TABLE:phy:0:lanes:200": {
+    "type": "hash",
+    "value": {
+      "index": "200",
+      "line_rx_lanemap": "0",
+      "line_to_system_lanemap": "0",
+      "line_tx_lanemap": "0",
+      "local_lane_id": "0",
+      "mdio_addr": "0x0200",
+      "rx_polarity": "0",
+      "system_side": "true",
+      "tx_polarity": "0"
+    }
+  }
+}
+~~~
+
+#### 3.4.2.4 GEARBOX_TABLE - Platform PHY Ports
+
+REDIS Example:
+~~~
+redis-dump -d 0 -k "_GEARBOX_TABLE:phy:0:ports:49" -y
+{
+  "_GEARBOX_TABLE:phy:0:ports:49": {
+    "type": "hash",
+    "value": {
+      "index": "49",
+      "line_adver_asym_pause": "false",
+      "line_adver_auto_neg": "false",
+      "line_adver_fec": "",
+      "line_adver_media_type": "fiber",
+      "line_adver_speed": "",
+      "line_auto_neg": "true",
+      "line_fec": "none",
+      "line_intf_type": "none",
+      "line_loopback": "none",
+      "line_media_type": "fiber",
+      "line_speed": "50000",
+      "line_training": "false",
+      "mdio_addr": "0x2000",
+      "system_auto_neg": "true",
+      "system_fec": "none",
+      "system_loopback": "none",
+      "system_speed": "25000",
+      "system_training": "false"
+    }
+  }
+}
+~~~
+
+### 3.4.3 STATE DB
```

Comment:
Do we need to add a table in STATE DB  in case of something out of orchagent needs to know gearbox info? Or do we have another way to achieve that?

Reply:
Hello inspurSDN and stephenxs;
Your questions are well received and should be considered for future design and development. Firmware download is not currently supported and there is no 'state' requirements at this time. The current HLD is focused on the initial configuration of a Gearbox with an emphasis to provide an initial infrastructure for vSONiC. Thank you for your participation and concerns.